### PR TITLE
Create is_model_deployed function in model.py

### PR DESCRIPTION
**Pull request by PrasannaSubramani**
@dhana555std i have created a isModelDeployed Function in model.py, so  we can use this in version file to check wheather the model is deployed in particular repo or not

### CODE REVIEW

File: cicd/resources/model.py [hunk 1]
```diff
@@ -271,3 +271,22 @@ def finish_initial_load(self, source_id):
...
+        for item in repo_list:
+            if item['@name'] == self.repository_name:
+                # If it matches, extract the names of 'mdm:Universe'
+                if 'mdm:Universe' in item:
+                    for universe in item['mdm:Universe']:
+                        universe_names.append(universe['@name'])
+        if self.model_name in universe_names:
+            return "true"
+        else:
+            return "false"
```
Comment (dhana555std) [inline cicd/resources/model.py]: Can't we write this in a single line?

File: cicd/resources/model.py [hunk 2]
```diff
@@ -271,3 +271,22 @@ def finish_initial_load(self, source_id):
         if response.status_code != 200:
             logger.info(f'Response is {response.content}')
             raise RuntimeError('Response is not 200. Exiting')
+
+    def is_model_deployed(self):
```
Comment (dhana555std) [inline cicd/resources/model.py]: I believe it is a good idea to pass the `repo name` or `repo id` as an argument here. Why are we trying to loop through all the repos?

File: cicd/resources/model.py [hunk 3]
```diff
@@ -271,3 +271,22 @@ def finish_initial_load(self, source_id):
...
+        url = f'{self.endpoint_url}/{self.account_id}/repositories'
+        response = requests.get(url=url, headers=self.headers)
+        dict_xml = (xmltodict.parse(response.content))
+        repo_list = dict_xml['mdm:Repositories']['mdm:Repository']
+        universe_names = []
```
Comment (dhana555std) [inline cicd/resources/model.py]: Can this be renamed to `models`?

**Review by dhana555std — CHANGES_REQUESTED** [review]

File: cicd/resources/model.py (hunk 2 above)
Comment (PrasannaSubramani) [reply cicd/resources/model.py]: I have passed the repo_id

File: cicd/resources/model.py [hunk 4]
```diff
@@ -271,3 +271,17 @@ def finish_initial_load(self, source_id):
         if response.status_code != 200:
             logger.info(f'Response is {response.content}')
             raise RuntimeError('Response is not 200. Exiting')
+
+    def is_model_deployed(self, repo_id):
```
Comment (dhana555std) [inline cicd/resources/model.py:275]: There is no documentation for this method.

File: cicd/resources/model.py [hunk 5]
```diff
@@ -271,3 +271,17 @@ def finish_initial_load(self, source_id):
...
+        repo_info = (xmltodict.parse(response.content))
+        models = []
+        model_info = repo_info['mdm:Repository']["mdm:Universe"]
+        if isinstance(model_info, list):
+            for item in model_info:
+                models.append(item['@name'])
+        else:
+            models.append(model_info["@name"])
+
+        return "true" if self.model_name in models else "false"
```
Comment (dhana555std) [inline cicd/resources/model.py]: Why are you returning true or false as a string? Why not a boolean?

File: cicd/resources/model.py [hunk 6]
```diff
@@ -271,3 +271,17 @@ def finish_initial_load(self, source_id):
...
+
+    def is_model_deployed(self, repo_id):
+        url = f'{self.endpoint_url}/{self.account_id}/repositories/{repo_id}'
+        response = requests.get(url=url, headers=self.headers)
+        repo_info = (xmltodict.parse(response.content))
+        models = []
+        model_info = repo_info['mdm:Repository']["mdm:Universe"]
+        if isinstance(model_info, list):
+            for item in model_info:
+                models.append(item['@name'])
```
Comment (dhana555std) [inline cicd/resources/model.py]: Can we use a map instead?

**Review by dhana555std — CHANGES_REQUESTED** [review]

**Review by dhana555std — APPROVED** [review]

**Pull request merged**